src/main/java/com/spartronics4915/frc2019/subsystems/MySubsystem.java

```diff
@@ -0,0 +1,28 @@
+public class MySubsystem extends Subsystem
+{
+    private static MySubsystem mInstance = null;
+
+    public static MySubsystem getInstance();
+    {
+        if (mInstance == null)
+        {
+            mInstance = new MySubsystem();
+        }
+        return mInstance;
+    }
+
```

Comment:
You need to add a private constructor for `MySubsystem` (the current implicit constructor is public). The way it is now, I can do `new MySubsystem`, which breaks the guarantee of the singleton pattern that we can only ever have one instance of `MySubsystem`.

E.g.:
```java
private MySubsystem()
{}
```

Reply:
i would be careful with more than a small number of singletons in our system.  The ones that seem justified:  Robot, Logger.

This is mostly because we've had a number of odd constructor timing issues. 
Since subsystems usually occur in single-instance patterns, it may be okay here.  That said, we must ensure that instances aren't created until RobotInit.  For this reason I prefer/recommend that all subsystems be member variables of Robot constructed via "mMySubsystem = new MySubsystem()" in robotInit.